bfvmm/include/hve/arch/intel_x64/lapic.h

```diff
@@ -120,15 +120,116 @@ class EXPORT_HYPERKERNEL_HVE lapic
     uint32_t base() const
     { return LAPIC_GPA; }
 
+    /// Read
+    ///
+    /// Read the value from a register
+    ///
+    /// @param idx the index of the register to read
+    //
+    /// @note the index is a dword offset, not a byte offset
+    ///
+    uint32_t read(uint32_t idx) const
+    { return m_lapic_view[idx]; }
+
+    /// Write
+    ///
+    /// Write the value to a register
+    ///
+    /// @param idx the index of the register to write
+    /// @param val the value to write
+    ///
+    /// @note the index is a dword offset, not a byte offset
+    ///
+    void write(uint32_t idx, uint32_t val)
+    { m_lapic_view[idx] = val; }
```

Comment:
Now that you have the x2APIC working again, is there a plan to delete all of the memory stuff? I would like to keep this to the bare minimum for now

Reply:
x2 doesn't work with PVH right now. specifically we would have to disable hvm_pvclock and hvm_pirqs

Reply:
☹️